homeassistant/components/nice_go/coordinator.py

```diff
@@ -178,16 +191,30 @@ async def _update_refresh_token(self) -> None:
 
     async def client_listen(self) -> None:
         """Listen to the websocket for updates."""
-        self.api.event(self.on_connected)
-        self.api.event(self.on_data)
-        try:
-            await self.api.connect(reconnect=True)
-        except ApiError:
-            _LOGGER.exception("API error")
+        self._unsub_connected = self.api.listen("on_connected", self.on_connected)
+        self._unsub_data = self.api.listen("on_data", self.on_data)
+        self._unsub_connection_lost = self.api.listen(
+            "on_connection_lost", self.on_connection_lost
+        )
+
+        for _ in range(RECONNECT_ATTEMPTS):
+            if self._hass_stopping:
```

Comment:
Why don't we just ask `self.hass.is_stopping`?

Reply:
`hass.is_stopping` is not set directly after the event is send. To synchronize testing it seems better to subscribe to the event atm.